lib/sshkit/host.rb

```diff
@@ -94,6 +78,13 @@ def properties
       @properties ||= OpenStruct.new
     end
 
+    def first_suitable_parser(host)
+      unless (parser = PARSERS.find{|p| p.suitable?(host) })
+        fail UnparsableHostStringError, "Cannot parse host string #{host}"
+      end
```

Comment:
I think this logic would be easier to understand if the `parser` assignment was broken out onto its own line instead of being within the `unless` condition. What do you think of this?

```ruby
parser = PARSERS.find { |p| p.suitable?(host) }
fail UnparsableHostStringError, "Cannot parse host string #{host}" if parser.nil?
```